### PR TITLE
[WIP] ElasticSearch implementation

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] Create index mapping on model creation
- [x] Index on record creation
- [x] Search view, proxy on E/S
- [x] Search with request body
- [x] Search with URL parameters
- [x] Reindex on record update
- [x] Index individual records on model records PUT
- [x] Test model definition update
- [x] Test model definition mapping (shapes, etc.)
- [x] Test model record mapping (object, etc.)
- [x] Test spatial query
- [x] Return specific response code when _elasticsearch_ not available/reachable

### CODE REVIEW

File: daybed/__init__.py [hunk 1]
```diff
@@ -141,6 +142,57 @@ def add_db_to_request(event):
         event.request.db = config.registry.backend.db()
     config.add_subscriber(add_db_to_request, NewRequest)
 
+    # index initialisation
```
Comment (almet) [inline daybed/__init__.py]: should we split this out in a separate function?

File: daybed/views/models.py [hunk 2]
```diff
@@ -42,6 +43,9 @@ def post_models(request):
         roles=request.validated['roles'],
         policy_id=request.validated['policy'])
 
+    event = ModelCreated(model_id, request)
+    request.registry.notify(event)
```
Comment (almet) [inline daybed/views/models.py]: what about having a `request.trigger_event` (or`notify`) utility function, which would have an `event_type` parameter?

We could then do something like `request.trigger_event("model_created", model_id, request)`.

(I find the current form to be a bit weird (using request.registry sound borked to me)

File: daybed/__init__.py (hunk 1 above)
Comment (leplatrem) [inline daybed/__init__.py]: yes of course !

File: daybed/views/models.py (hunk 2 above)
Comment (leplatrem) [inline daybed/views/models.py]: To be honest, it's the first time I use Pyramid events :) Yes, why not using Javascript-like events !

Comment (leplatrem) [inline daybed/views/models.py]: Note: http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/events.html#creating-your-own-events

(helper would still be relevant)

**Pull request closed without merging**